extensions-contrib/rocketmq-indexing-service/src/main/java/org/apache/druid/indexing/rocketmq/RocketMQRecordSupplier.java

```diff
@@ -0,0 +1,243 @@
+        consumer.seekToEnd(mq);
+      }
+      catch (MQClientException e) {
+        log.error(e.getErrorMessage());
+      }
+    }
+  }
+
+  @Override
+  public Set<StreamPartition<String>> getAssignment()
+  {
+    if (this.streamPartitions == null) {
+      return new HashSet<StreamPartition<String>>();
+    }
+    return this.streamPartitions;
+  }
+
+  @Nonnull
+  @Override
+  public List<OrderedPartitionableRecord<String, Long, RocketMQRecordEntity>> poll(long timeout)
+  {
+    List<OrderedPartitionableRecord<String, Long, RocketMQRecordEntity>> polledRecords = new ArrayList<>();
+    for (MessageExt messageExt : consumer.poll(timeout)) {
+
+      polledRecords.add(new OrderedPartitionableRecord<>(
+          messageExt.getTopic(),
+          PartitionUtil.genPartition(messageExt.getBrokerName(), messageExt.getQueueId()),
+          messageExt.getQueueOffset(),
+          messageExt.getBody() == null ? null : ImmutableList.of(new RocketMQRecordEntity(messageExt.getBody()))
+      ));
+    }
+    return polledRecords;
+  }
+
+  @Override
+  public Long getLatestSequenceNumber(StreamPartition<String> partition)
+  {
+    long offset = Long.MAX_VALUE;
+    try {
+      offset = clientInstance.getMQAdminImpl().maxOffset(PartitionUtil.genMessageQueue(partition.getStream(), partition.getPartitionId()));
+    }
+    catch (MQClientException e) {
+      log.error(e.getErrorMessage());
+    }
+    return offset;
+  }
+
+  @Override
+  public Long getEarliestSequenceNumber(StreamPartition<String> partition)
+  {
+    long offset = 0;
+    try {
+      offset = clientInstance.getMQAdminImpl().minOffset(PartitionUtil.genMessageQueue(partition.getStream(), partition.getPartitionId()));
+    }
+    catch (MQClientException e) {
+      log.error(e.getErrorMessage());
+    }
+    return offset;
+  }
+
+  @Nullable
+  @Override
+  public Long getPosition(StreamPartition<String> partition)
+  {
+    MessageQueue mq = PartitionUtil.genMessageQueue(partition.getStream(), partition.getPartitionId());
+    return consumer.getOffsetStore().readOffset(mq, ReadOffsetType.READ_FROM_STORE);
+  }
+
+  @Override
+  public Set<String> getPartitionIds(String stream)
+  {
+    HashSet<String> partitionIds = new HashSet<>();
+    Collection<MessageQueue> messageQueues = null;
+    try {
+      if (!consumer.isRunning()) {
+        consumer.start();
+      }
+      messageQueues = consumer.fetchMessageQueues(stream);
+    }
+    catch (MQClientException e) {
+      log.error(e.getErrorMessage());
+    }
+    if (messageQueues != null) {
+      for (MessageQueue messageQueue : messageQueues) {
+        partitionIds.add(PartitionUtil.genPartition(messageQueue));
+      }
+    }
+    return partitionIds;
+  }
+
+  @Override
+  public void close()
+  {
+    if (closed) {
+      return;
+    }
+    closed = true;
+    consumer.shutdown();
+  }
+
+  private static DefaultLitePullConsumer getRocketMQConsumer(ObjectMapper sortingMapper, Map<String, Object> consumerProperties)
```

Comment:
first arg is not used inside this function

Reply:
Deleted it.